src/codeu/chat/client/LoginDialog.java

```diff
@@ -0,0 +1,125 @@
+package codeu.chat.client;
+
+import java.awt.*;
+import java.awt.event.*;
+import javax.swing.*;
+import javax.swing.border.*;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import codeu.chat.common.BasicView;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.ConversationSummary;
+import codeu.chat.common.LogicalView;
+import codeu.chat.common.Message;
+import codeu.chat.common.NetworkCode;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.connections.ConnectionSource;
+import codeu.chat.client.simplegui.UserPanel;
+
+public class LoginDialog extends JDialog {
+ 
+    private JTextField tfUsername;
+    private JPasswordField pfPassword;
+    private JLabel lbUsername;
+    private JLabel lbPassword;
+    private JButton btnLogin;
+    private JButton btnCancel;
+    private boolean succeeded;
+ 
+    public LoginDialog(Frame parent) {
+        super(parent, "Login", true);
+        //
+        JPanel panel = new JPanel(new GridBagLayout());
+        GridBagConstraints cs = new GridBagConstraints();
```

Comment:
I don't like this variable name but I don't know if this is standard for Swing stuff.